packages/grafana-ui/src/components/uPlot/config.ts

```diff
@@ -93,7 +93,7 @@ export interface LineConfig {
   lineWidth?: number;
   lineInterpolation?: LineInterpolation;
   lineStyle?: LineStyle;
-  spanNulls?: boolean;
+  spanNulls?: boolean | number;
```

Comment:
Thinking more... should our setting be "boolean | string"?

The string should parse values like "1s", "5m", "10s" -- that maps to the existing settings for things like "min interval"

Reply:
i think this needs to be a unitless value that applies to whatever x field happens to be, which may not be time or which may be 10ms.

Reply:
Is the goal to allow arbitrary durations in multiples of 1s? If not - we only want to allow a finite set of durations - then string is obviously preferable.

If so, and this were greenfield, i'd say `number` is preferable. Better to just use seconds as a standard, and treat it as a higher layer's responsibility to translate those seconds to a more convenient human-readable form.

But if the `string`-based pattern is already established elsewhere AND "maps to" here:

> that maps to the existing settings for things like "min interval"

means there's circumstances where we want to take these values and copy/combine them with those fields, then number isn't enough better to justify the divergence.

Reply:
> Is the goal to allow arbitrary durations in multiples of 1s?

well, with streaming we're aiming for 20hz, so 50ms resolution. and the time axis supports resolutions down to 1ms. i imagine restricting to some predefined values will be pretty limiting for various data resolutions people may have. and a number also covers cases where x is not time.

i'm still new here, so 🤷 :)

Reply:
@torkelo -- I think there used to be a frontend equivalent to [time.ParseDuration](https://golang.org/pkg/time/#ParseDuration), but I can't find it.  there is datemath.parse, but that is slightly different

Reply:
I found it... 
```
 const interval = rangeUtil.intervalToSeconds( strvalue );
    ```

Reply:
no issue with string, as long as we can parse & specify arbitrary values. my main concern is not setting arbitrary limits on what interval can be configured here.

no real reason not to support `boolean | number | string`. the main goal of this PR was to add an api that's maximally flexible and Explore can configure arbitrary values when using this component. the string variant can br added later when / if we choose to expose this in the UI somehow?

another point is i think this will be the first option that has to be parsed in the config at this level?

Reply:
> as long as we can parse & specify arbitrary values

:100: yes

> no real reason not to support boolean | number | string

The only consideration is what is in the saved JSON vs what is runtime.  If we can simplify what is in JSON, i think that is best.  string covers everythign, so I lean towards that.

> i think this will be the first option that has to be parsed in the config at this level?

lots of things like this -- for graph, we have moved most of that further up the stack so you just have a function to get the color rather than needing to read what the color mode is, etc

Reply:
yeah that was mostly what i was wondering. should this gk up the stack, too?

Reply:
i don't think so... the other things that have moved further up have clear use across most all visualizations (ie, color, units, decimals, etc)

This pattern will be useful for a few -- but hard to think where it fits in a global setting

Reply:
> If we can simplify what is in JSON, i think that is best

💯 💯 💯